File: app/Http/Controllers/SpecialistPatternsJava/Creational/AbstractFactory/AbstractFactoryJava/src/factories/WoodHouseFactory.java
package factories;

import roofs.Roof;
import roofs.WoodRoof;
import walls.Wall;
import walls.WoodWall;
import windows.Window;
import windows.WoodFrameWindow;

// WoodHouseFactory - конкретная фабрика для создания компонентов деревянных домов.
public class WoodHouseFactory implements HouseFactory {

    // Метод создания стены, возвращает деревянную стену.
    @Override
    public Wall createWall() {
        return new WoodWall();
    }

    // Метод создания крыши, возвращает крышу с деревянной отделкой.
    @Override
    public Roof createRoof() {
        return new WoodRoof();
    }

    // Метод создания окна, возвращает окно с деревянной рамой.
    @Override
    public Window createWindow() {
        return new WoodFrameWindow();
    }
}

File: app/Http/Controllers/SpecialistPatterns/AbstractFactory/Factories/WoodHouseFactory.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\AbstractFactory\Factories;

use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Roofs\Roof;
use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Roofs\WoodRoof;
use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Walls\Wall;
use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Walls\WoodWall;
use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Windows\Window;
use App\Http\Controllers\SpecialistPatterns\AbstractFactory\Windows\WoodFrameWindow;
use JetBrains\PhpStorm\Pure;

// WoodHouseFactory - конкретная фабрика для создания компонентов деревянных домов.
class WoodHouseFactory implements HouseFactory
{

    // Метод создания стены, возвращает деревянную стену.
    #[Pure] public function createWall(): Wall
    {
        return new WoodWall();
    }

    // Метод создания крыши, возвращает крышу с деревянной отделкой.
    #[Pure] public function createRoof(): Roof
    {
        return new WoodRoof();
    }

    // Метод создания окна, возвращает окно с деревянной рамой.
    #[Pure] public function createWindow(): Window
    {
        return new WoodFrameWindow();
    }
}
